pkg/serviceaccount/util.go

```diff
@@ -79,3 +88,90 @@ func IsServiceAccountToken(secret *v1.Secret, sa *v1.ServiceAccount) bool {
 
 	return true
 }
+
+func PublicJWKSFromKeys(in []interface{}) (*jose.JSONWebKeySet, errors.Aggregate) {
+	// Decode keys into a JWKS.
+	var keys jose.JSONWebKeySet
+	var errs []error
+	for i, key := range in {
+		var pubkey *jose.JSONWebKey
+		var err error
+
+		switch k := key.(type) {
+		case interface {
+			Public() crypto.PublicKey
+		}:
+			// This is a private key. Get its public key
+			pubkey, err = jwkFromPublicKey(k.Public())
+		case rsa.PublicKey:
```

Comment:
Only accept pointers here. We don't accept structs lower and neither does the standard lib.

Reply:
That looks suboptimal. I think we should prefer:

```
pk := sk.(crypto.Signer).Public()
```

I'm hesitant to be more permissive than the stdlib without a reason.

https://github.com/golang/go/blob/22d28a24c8b0d99f2ad6da5fe680fa3cfa216651/src/crypto/x509/x509.go#L71-L73

Reply:
Hmm, if I do that, a bunch of other test cases fail with `invalid bearer token, square/go-jose: unsupported key type/format`.

Reply:
On the rejecting non-pointers though, I agree since go-jose does the same thing: https://github.com/kubernetes/kubernetes/blob/master/vendor/gopkg.in/square/go-jose.v2/signing.go#L132-L160

Reply:
Weird, I get the same error if I just do `pk := &sk.(*ecdsa.PrivateKey).PublicKey` to make it a pointer...

Reply:
Remove the & here:

https://github.com/kubernetes/kubernetes/blob/5a45e300819b6d3248fed620ba07a893cf448908/test/integration/auth/svcaccttoken_test.go#L86

Reply:
Derp, that was it